.github/actions/javascript/authorChecklist/authorChecklist.ts

```diff
@@ -0,0 +1,177 @@
+import core from '@actions/core';
+import github from '@actions/github';
+import https from 'https';
+import _ from 'lodash';
+import GithubUtils from '../../../libs/GithubUtils';
+import CONST from '../../../libs/CONST';
+import newComponentCategory from './newComponentCategory';
+
+const pathToAuthorChecklist = 'https://raw.githubusercontent.com/Expensify/App/main/.github/PULL_REQUEST_TEMPLATE.md';
+const checklistStartsWith = '### PR Author Checklist';
+const checklistEndsWith = "\r\n### Screenshots/Videos";
+
+const prNumber = github.context.payload.pull_request?.number;
+
+const CHECKLIST_CATEGORIES = {
+    NEW_COMPONENT: newComponentCategory,
+};
+
+/**
+ * Look at the contents of the pull request, and determine which checklist categories apply.
+ */
+async function getChecklistCategoriesForPullRequest() {
+    const categories = [];
+    const changedFiles = await GithubUtils.paginate(GithubUtils.octokit.pulls.listFiles, {
+        owner: CONST.GITHUB_OWNER,
+        repo: CONST.APP_REPO,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        pull_number: prNumber,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        per_page: 100,
+    });
+
+    for (const category of Object.values(CHECKLIST_CATEGORIES)) {
+        const {detectFunction, items} = category;
+        const categoryDetected = await detectFunction(changedFiles);
+        if (categoryDetected) {
+            categories.push(items);
```

Comment:
I'm getting type errors here 🤔 

I think we should add type to categories definition:
```ts
const categories: string[][] = [];
```